src/overlays/actors/ovl_Demo_Gj/z_demo_gj.c

```diff
@@ -96,241 +84,1326 @@ static ColliderCylinderInitType1 sCylinderInit = {
+    DemoGj_Cylinder_Init(this, globalCtx, &this->cylinders[2], &sCylinderInit2);
+}
+
+void func_8097B444(DemoGj* this, GlobalContext* globalCtx) {
+}
+
+void func_8097B450(DemoGj* this, GlobalContext* globalCtx) {
+    ColliderCylinder* cylinder0 = &this->cylinders[0];
+    ColliderCylinder* cylinder1 = &this->cylinders[1];
+    ColliderCylinder* cylinder2 = &this->cylinders[2];
+    PosRot* actorPosRot = &this->dyna.actor.world;
+    s32 pad;
+    s16 theta = actorPosRot->rot.y;
+    f32 cos_theta = Math_CosS(theta);
+    f32 sin_theta = Math_SinS(theta);
+
+    cylinder0->dim.pos.z = actorPosRot->pos.z - (35.0f * sin_theta);
+    cylinder0->dim.pos.x = actorPosRot->pos.x + (35.0f * cos_theta);
+    cylinder0->dim.pos.y = actorPosRot->pos.y;
+
+    cylinder1->dim.pos.z = actorPosRot->pos.z - (-10.0f * sin_theta);
+    cylinder1->dim.pos.x = actorPosRot->pos.x + (-10.0f * cos_theta);
+    cylinder1->dim.pos.y = actorPosRot->pos.y;
+
+    cylinder2->dim.pos.z = actorPosRot->pos.z - (-55.0f * sin_theta);
+    cylinder2->dim.pos.x = actorPosRot->pos.x + (-55.0f * cos_theta);
+    cylinder2->dim.pos.y = actorPosRot->pos.y;
+}
+
+void func_8097B5A4(DemoGj* this, GlobalContext* globalCtx2) {
+    GlobalContext* globalCtx = globalCtx2;
+    CollisionCheckContext* colChkCtx = &globalCtx->colChkCtx;
+    Collider* cylinders_0 = &this->cylinders[0].base;
+    Collider* cylinders_1 = &this->cylinders[1].base;
+    Collider* cylinders_2 = &this->cylinders[2].base;
+    s32 pad[3];
+
+    CollisionCheck_SetAC(globalCtx, colChkCtx, cylinders_0);
+    CollisionCheck_SetAC(globalCtx, colChkCtx, cylinders_1);
+    CollisionCheck_SetAC(globalCtx, colChkCtx, cylinders_2);
+}
+
+// Does the same as `DemoGj_Cylinder_AnyHasExploded`
+s32 DemoGj_Cylinder_AnyHasExploded2(DemoGj* this, GlobalContext* globalCtx) {
+    if (DemoGj_Cylinder_HasExploded(this, globalCtx, &this->cylinders[0])) {
+        return 1; // return true; (?)
+    }
+    if (DemoGj_Cylinder_HasExploded(this, globalCtx, &this->cylinders[1])) {
+        return 1; // return true; (?)
+    }
+    if (DemoGj_Cylinder_HasExploded(this, globalCtx, &this->cylinders[2])) {
+        return 1; // return true; (?)
+    }
+    return 0; // return false; (?)
+}
+
+void func_8097B688(DemoGj* this, GlobalContext* globalCtx, Vec3f* arg2) {
+    Vec3f pos;
+
+    pos.x = this->dyna.actor.world.pos.x;
+    pos.y = this->dyna.actor.world.pos.y;
+    pos.z = this->dyna.actor.world.pos.z;
+
+    func_80978C20(this, globalCtx, &pos, arg2);
+}
+
+void func_8097B6C4(DemoGj* this, GlobalContext* globalCtx) {
+    Vec3f* scale = &this->dyna.actor.scale;
+
+    if (func_8097983C(this, globalCtx)) {
+        DemoGj_InitCommon(this, globalCtx, &D_06002448);
+        this->updateIndex = 19;
+        this->drawIndex = 17;
+        scale->x *= 0.8f;
+        scale->y *= 0.8f;
+        scale->z *= 0.8f;
+    }
+}
+
+static Vec3f D_8097BEB8 = { 0.0f, 0.0f, 0.0f };
+
```

Comment:
this vector can be moved inside, and you can even make it non-static and remove the temp. just do vec1 = {};